src/track/beatgrid.cpp

```diff
@@ -98,7 +98,8 @@ BeatsPointer BeatGrid::makeBeatGrid(
     // Calculate beat length as sample offsets
     const audio::FrameDiff_t beatLengthFrames = 60.0 * sampleRate / bpm.value();
 
-    return BeatsPointer(new BeatGrid(sampleRate, subVersion, grid, beatLengthFrames));
+    return std::make_shared<BeatGrid>(
+            MakeSharedTag{}, sampleRate, subVersion, grid, beatLengthFrames);
```

Comment:
Why do you use here the explicit version, where above you use only the braced tantalizer?

Reply:
Because std::make_shared() is not able to deduce the type and compilation fails. Try it.

Reply:
I think we should be explicit everywhere.